NeuroID/src/main/java/com/neuroid/tracker/storage/NIDDataStoreManager.kt

```diff
@@ -43,17 +47,31 @@ private object NIDDataStoreManagerImp : NIDDataStoreManager {
     private const val NID_STRING_JSON_PAYLOAD = "NID_STRING_JSON_PAYLOAD"
     private var sharedPref: SharedPreferences? = null
     private val listNonActiveEvents = listOf(
-        USER_INACTIVE,
-        WINDOW_BLUR //Block screen
+        USER_INACTIVE, WINDOW_BLUR //Block screen
     )
     private val listIdsExcluded = arrayListOf<String>()
+    private var queuedEvents: Queue<NIDEventModel> = LinkedList()
 
     fun init(context: Context) {
         CoroutineScope(Dispatchers.IO).launch {
             sharedPref = context.getSharedPreferences(NID_SHARED_PREF_FILE, MODE_PRIVATE)
         }
     }
 
+    // Queue events that are set before sdk is started
+    @Synchronized
+    override fun queueEvent(event: NIDEventModel) {
+        queuedEvents.add(event)
+    }
+
+    @Synchronized
+    override fun saveAndClearAllQueuedEvents() {
+        if (queuedEvents.isNotEmpty()) {
+            queuedEvents.forEach { event -> getDataStoreInstance().saveEvent(event) }
+            queuedEvents.clear()
+        }
+    }
+
```

Comment:
Just a style comment, no need to change, but the check for `isNotEmpty` is unnecessary.

Reply:
Got it. Since the collection is initialized it won't throw a null error and can be iterated even if empty. Will remove that in a following PR.